src/symbolize/gimli/libs_windows.rs

```diff
@@ -43,13 +42,67 @@ unsafe fn add_loaded_images(ret: &mut Vec<Library>) {
     }
 }
 
+// Safety: long_path should be null-terminated
+#[cfg(target_os = "cygwin")]
+unsafe fn get_posix_path(long_path: &[u16]) -> Option<OsString> {
+    use super::mystd::os::unix::ffi::OsStringExt;
+
+    unsafe extern "C" {
+        fn cygwin_conv_path(
+            what: libc::c_uint,
+            from: *const libc::c_void,
+            to: *mut libc::c_void,
+            size: libc::size_t,
+        ) -> libc::ssize_t;
+    }
+    const CCP_WIN_W_TO_POSIX: libc::c_uint = 3;
+
+    let name_len = unsafe {
+        cygwin_conv_path(
+            CCP_WIN_W_TO_POSIX,
+            long_path.as_ptr().cast(),
+            core::ptr::null_mut(),
+            0,
+        )
+    };
+    // Expect at least 1 for null terminator.
+    if name_len < 1 {
+        return None;
+    }
```

Comment:
If it's -1 then it's an error condition. Should we check errno and react to that?

Reply:
If the answer is "no" then we should probably comment that we are explicitly saying "lol" to any error.

Reply:
I checked the documentation and found that it's not likely to return error here. I think just ignoring it is fine.